src/apply_rules.rs

```diff
@@ -82,6 +96,91 @@ fn read_complementary_code_rules_files<P: AsRef<Path>>(
     Ok(codes.into_iter().collect())
 }
 
+fn read_property_rules_files<P: AsRef<Path>>(
+    rule_files: Vec<P>,
+    report: &mut Report,
+) -> Result<Vec<PropertyRule>> {
+    info!("Reading property rules.");
+    let mut properties: BTreeMap<(ObjectType, String, String), BTreeSet<PropertyRule>> =
+        BTreeMap::default();
+    for rule_path in rule_files {
+        let path = rule_path.as_ref();
+        let mut rdr = csv::ReaderBuilder::new()
+            .trim(csv::Trim::All)
+            .from_path(&path)
+            .with_context(ctx_from_path!(path))?;
+        for p in rdr.deserialize() {
+            let p: PropertyRule = match p {
+                Ok(val) => val,
+                Err(e) => {
+                    report.add_warning(
+                        format!("Error reading {:?}: {}", path.file_name().unwrap(), e),
+                        ReportType::PropertyRulesRead,
+                    );
+                    continue;
+                }
+            };
+
+            properties
+                .entry((p.object_type, p.object_id.clone(), p.property_name.clone()))
+                .or_insert_with(BTreeSet::new)
+                .insert(p);
+        }
+    }
+
+    let properties = properties
+        .iter()
+        .filter(|((object_type, object_id, property_name), property)| {
+            match object_type {
+                ObjectType::Route
+                    if ![
+                        "route_name",
```

Comment:
not really sure, (I don't know exactly the features you want), but shouldn't stuff like this be handled at serialization (maybe en enum with different kind of rules ?) 

if not handled by the type system, it might be worth it to put this inside a `is_valid` method

@TeXitoi what do you think ?

Reply:
Using an enum would be better.

Reply:
Mmm, enum is complicated as it depends on the object_type. But better to abstract that somewhere.